token/cli/src/main.rs

```diff
@@ -5810,6 +5839,251 @@ mod tests {
         assert!(account.is_err());
     }
 
+    #[tokio::test]
+    #[serial]
+    async fn burn_with_permanent_delegate() {
+        let (test_validator, payer) = new_validator_for_test().await;
+        let config =
+            test_config_with_default_signer(&test_validator, &payer, &spl_token_2022::id());
+
+        let token_keypair = Keypair::new();
+        let token = token_keypair.pubkey();
+        let bulk_signers: Vec<Arc<dyn Signer>> =
+            vec![Arc::new(clone_keypair(&payer)), Arc::new(token_keypair)];
+
+        command_create_token(
+            &config,
+            TEST_DECIMALS,
+            token,
+            payer.pubkey(),
+            false,
+            false,
+            false,
+            true,
+            None,
+            None,
+            None,
+            None,
+            bulk_signers,
+        )
+        .await
+        .unwrap();
+
+        let source = Keypair::new();
+
+        let permanent_delegate_keypair_file = NamedTempFile::new().unwrap();
+        write_keypair_file(&payer, &permanent_delegate_keypair_file).unwrap();
+
+        let source_keypair_file = NamedTempFile::new().unwrap();
+        write_keypair_file(&source, &source_keypair_file).unwrap();
```

Comment:
nit: Up to you, but I think you can use the `create_associated_account` helper rather than going through the whole keypair saving annoyance, unless I'm missing something